data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/accumulator/LocalFileBufferFactoryTest.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.accumulator;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+class LocalFileBufferFactoryTest {
+    @Test
+    void test_localFileBufferFactory_notNull() {
+        LocalFileBufferFactory localFileBufferFactory = new LocalFileBufferFactory();
+        Assertions.assertNotNull(localFileBufferFactory);
+    }
+
+    @Test
+    void test_buffer_notNull() {
+        LocalFileBufferFactory localFileBufferFactory = new LocalFileBufferFactory();
+        Assertions.assertNotNull(localFileBufferFactory);
+        Buffer buffer = localFileBufferFactory.getBuffer();
+        Assertions.assertNotNull(buffer);
+        assertThat(buffer, instanceOf(Buffer.class));
```

Comment:
You should assert that this is an instance of `LocalFileBuffer` as well.

```
assertThat(buffer, instanceOf(LocalFileBuffer.class))
```

Reply:
Addressed.

Reply:
@dlvenable, Earlier, I didn't understand. Now I can understand and modify as requested.

Reply:
It should be as simple as changing this line to:

```
assertThat(buffer, instanceOf(LocalFileBuffer.class))
```

Here it is with a little more context.

```
Buffer buffer = localFileBufferFactory.getBuffer();
Assertions.assertNotNull(buffer);
assertThat(buffer, instanceOf(LocalFileBuffer.class))
```

We already know this is a `Buffer`. That is handled statically by Java. But, we want to know that this is returning the `LocalFileBuffer` and not an `InMemoryBuffer` (or some other buffer sub-class).

Reply:
Thanks for your clarification, Modified as you suggested.